Modular Exponentiation/modularExponentiation.cpp

```diff
@@ -0,0 +1,35 @@
+#include <bits/stdc++.h>
+using namespace std;
+#define ld long double
+#define ulint unsigned long long int
+#define lint long long int
+#define mp make_pair
+#define pb push_back
+#define f(x) for(lint i=0;i<x;i++)
+#define fj(x) for(lint j=0;j<x;j++)
+#define F first
+#define S second
+#define faster ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0)
+
+lint modex(lint base,lint exponent,lint mod){
+    if(base == 0){
+        return 0;		
+    }
+    if(exponent == 0){
+        return 1;
+    }
+    if(exponent%2 == 0){
+        lint ans = modex(base,exponent/2,mod);
+        return (ans*ans)%mod;
```

Comment:
What if the user puts base = 10^10? ans * ans will overflow here. You should use expanded modular here.

Reply:
if the user puts base as 10^10, firstly there will be a recursive call until the exponent gets odd, and when it does, it will return the value which is already % 10^9+7. So, ans will never be > mod. hence, ans*ans will not overflow.

Reply:
ahh, yes sorry for that. I missed it there.